cognite/client/utils/_time.py

```diff
@@ -178,6 +178,23 @@ def datetime_to_ms_iso_timestamp(dt: datetime) -> str:
     raise TypeError(f"Expected datetime object, got {type(dt)}")
 
 
+def timestamp_str_to_datetime(timestamp: str) -> datetime:
+    """Converts a timestamp string in ISO 8601 format to a datetime object.
+
+    Args:
+        timestamp (str): Timestamp string in ISO 8601 format
+
+    Returns:
+        datetime: Datetime object
+    """
+    try:
+        return datetime.fromisoformat(timestamp)
+    except ValueError:
+        # Typically hits if the timestamp is missing milliseconds
+        # For example, "2021-01-01T00:00:00.17+00:00", i.e. missing the last digit
+        return datetime.strptime(timestamp, "%Y-%m-%dT%H:%M:%S.%f%z")
```

Comment:
How about we remove this function and just use this `strptime` directly which seem to work in all cases?